flink/src/main/java/org/apache/iceberg/flink/source/FlinkInputFormat.java

```diff
@@ -91,11 +105,85 @@ public void configure(Configuration parameters) {
 
   @Override
   public void open(FlinkInputSplit split) {
+    boolean enableVectorizedRead = readableConfig.get(FlinkTableOptions.ENABLE_VECTORIZED_READ);
+
+    if (enableVectorizedRead) {
+      if (useOrcVectorizedRead(split.getTask())) {
```

Comment:
I think this check should be moved inside `BatchRowDateIterator`. Here the logic should be simple 
```
if (enableVectorizedRead) {
  ... new BatchRowDataIterator
} else {
   ... new RowDataIterator
}

Reply:
We also don't need the separate method of `setDefaultIterator`

Reply:
> I think this check should be moved inside `BatchRowDateIterator`. Here the logic should be simple
> 
> ```
> if (enableVectorizedRead) {
>   ... new BatchRowDataIterator
> } else {
>    ... new RowDataIterator
> }
> ```

done

Reply:
> We also don't need the separate method of `setDefaultIterator`

At first, I wanted to use non-vectorized read when  judge failed of vectorized read, so I extracted this method, but later I think it inappropriate.